### PR TITLE
Fix loading user-defined icons for Tk toolbar

### LINKED ISSUE #22131: [Bug]: png icon image fails to load for manually defined tool buttons
### Bug summary

The ToolBase class has the interface of overriding the property 'image' containing the path to an image file to be displayed as the icon of the tool button:
https://matplotlib.org/stable/api/backend_tools_api.html#matplotlib.backend_tools.ToolBase.image
This image fails to load on Qt5 backends for PNG image files.

### Code for reproduction

```python
import matplotlib
matplotlib.use("Qt5Agg")
matplotlib.rcParams["toolbar"] = "toolmanager"
from matplotlib.backend_tools import ToolToggleBase
import matplotlib.pyplot as plt

class ExampleToolToggle(ToolToggleBase):
    image = "/tmp/example.png"

figure, axis = plt.subplots()
tool_manager = figure.canvas.manager.toolmanager
tool_manager.add_tool("example", ExampleToolToggle)
toolbar = figure.canvas.manager.toolbar
toolbar.add_tool(tool_manager.get_tool("example"), "zoompan", 0)
plt.show()
```

### Actual outcome

The name of the tool is displayed instead of the image in the toolbar.

### Expected outcome

The image displayed in the toolbar.

### Additional information

This bug was introduced in version 3.3.0 (3.2.2 works fine). 
The problem is caused by ToolbarQt calling the _icon method of NavigationToolbar2QT at:
https://github.com/matplotlib/matplotlib/blob/710fce3df95e22701bd68bf6af2c8adbc9d67a79/lib/matplotlib/backends/backend_qt.py#L924

The _icon method unconditionally changes the path by inserting '_large' before the '.png' suffix:
https://github.com/matplotlib/matplotlib/blob/710fce3df95e22701bd68bf6af2c8adbc9d67a79/lib/matplotlib/backends/backend_qt.py#L680

Before version 3.3.0 a _icon method of ToolbarQt itself was called, which didn't have the path mangling.

### Operating system

Ubuntu

### Matplotlib Version

3.5.1

### Matplotlib Backend

Qt5Agg

### Python version

Python 3.8.12

### Jupyter version

_No response_

### Installation

pip

### PR DESCRIPTION
**Pull request by daniilS**
## PR Summary
Tk equivalent of #22135, closes #22131

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/backends/_backend_tk.py [hunk 1]
```diff
@@ -643,9 +643,15 @@ def _set_image_for_button(self, button):
         if button._image_file is None:
             return
 
+        # Allow _image_file to be relative to Matplotlib's "images" data
+        # directory.
+        path_regular = cbook._get_data_path('images', button._image_file)
+        path_large = cbook._get_data_path(
+            'images', button._image_file.replace('.png', '_large.png'))
```
Comment (QuLogic) [inline lib/matplotlib/backends/_backend_tk.py:650]:
```suggestion
        path_large = path_regular.with_name(
            button._image_file.replace('.png', '_large.png'))
```

Comment (daniilS) [reply lib/matplotlib/backends/_backend_tk.py:650]: Thanks!

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
@timhoffm Why is this one 3.6.0 only?

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.5.x

**Pull request merged**